rules/aip0123/resource_type_name.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0123
+
+import (
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var resourceTypeName = &lint.MessageRule{
+	Name: lint.NewRuleName(123, "resource-type-name"),
+	OnlyIf: func(m *desc.MessageDescriptor) bool {
+		return utils.GetResource(m) != nil
+	},
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		resource := utils.GetResource(m)
+		if strings.Count(resource.GetType(), "/") != 1 {
+			return []lint.Problem{{
+				Message:    "Resource type names must be of the form {Service Name}/{Type}.",
+				Descriptor: m,
+				Location:   locations.MessageResource(m),
```

Comment:
I don't think so, I think the best we can do is point at the annotation itself. I can't tell if the source code info goes as deep as fields within an extension. Since there is only one `type` per `google.api.resource` pointing at the entire annotation should be enough? wdyt?